src/Exchange/ReversedCurrenciesExchange.php

```diff
@@ -11,34 +13,30 @@
  * Tries the reverse of the currency pair if one is not available.
  *
  * Note: adding nested ReversedCurrenciesExchange could cause a huge performance hit.
- *
- * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
 final class ReversedCurrenciesExchange implements Exchange
 {
-    /**
-     * @var Exchange
-     */
-    private $exchange;
+    private Exchange $exchange;
 
     public function __construct(Exchange $exchange)
     {
         $this->exchange = $exchange;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function quote(Currency $baseCurrency, Currency $counterCurrency)
+    public function quote(Currency $baseCurrency, Currency $counterCurrency): CurrencyPair
     {
         try {
             return $this->exchange->quote($baseCurrency, $counterCurrency);
         } catch (UnresolvableCurrencyPairException $exception) {
             try {
                 $currencyPair = $this->exchange->quote($counterCurrency, $baseCurrency);
 
-                return new CurrencyPair($baseCurrency, $counterCurrency, 1 / $currencyPair->getConversionRatio());
-            } catch (UnresolvableCurrencyPairException $inversedException) {
+                return new CurrencyPair(
+                    $baseCurrency,
+                    $counterCurrency,
+                    (string) (1.0 / (float) $currencyPair->getConversionRatio())
```

Comment:
I'd love to remove this float here, but I guess this would require the use of a calculator. Unless you see the opportunity to fix this in this PR, this should be a task after merging.

Reply:
Initially wanted to use the calculator here, but avoided introducing changes in components that are not performance sensitive. Most changes you see have been done to sustain the improvements in tests and performance of the core `Money` VO.

I would say new issue and defer here.

Reply:
Makes sense, see #636